core/env/rassoc_spec.rb

```diff
@@ -1,15 +1,30 @@
 require_relative '../../spec_helper'
 
 describe "ENV.rassoc" do
+  before :each do
+    @foo = ENV["foo"]
+    @baz = ENV["baz"]
+  end
+
   after :each do
-    ENV.delete("foo")
+    ENV["foo"] = @foo
+    ENV["baz"] = @baz
   end
 
   it "returns an array of the key and value of the environment variable with the given value" do
     ENV["foo"] = "bar"
     ENV.rassoc("bar").should == ["foo", "bar"]
   end
 
+  it "returns a single array even if there are multiple such environment variables" do
+    ENV["foo"] = "bar"
+    ENV["baz"] = "bar"
+    [
+        ["foo", "bar"],
+        ["baz", "bar"],
+    ].should include(ENV.rassoc("bar"))
```

Comment:
Can it ever change or is it always the same? I'd it's always the first one.

Reply:
"First one" is OS-dependent.  Alpha on Windows, not-alpha on Linux.   It can't change on one OS, but will be different on different OSes.  The test should be OS-independent.
```

Reply:
Right, it's up to the OS to choose the order of ENV vars, good point.